src/realm/parser/collection_operator_expression.hpp

```diff
@@ -53,33 +53,42 @@ struct CollectionOperatorExpression
         if (requires_suffix_path) {
             Table* pre_link_table = pe.table_getter();
             REALM_ASSERT(pre_link_table);
-            StringData list_property_name = pre_link_table->get_column_name(pe.col_ndx);
-            precondition(pe.col_type == type_LinkList,
+            StringData list_property_name;
+            if (pe.dest_type_is_backlink()) {
+                list_property_name = "linking object";
+            } else {
+                list_property_name = pre_link_table->get_column_name(pe.get_dest_ndx());
+            }
+            precondition(pe.get_dest_type() == type_LinkList || pe.dest_type_is_backlink(),
```

Comment:
this `precondition` macro is defined in a .hpp file (parser_utils.hpp), so it must be prefixed with REALM_ in order not to pollute users´ name spaces